src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -491,6 +491,9 @@ public IThreadPoolWorkItem Dequeue(ThreadPoolWorkQueueThreadLocals tl, ref bool
 
         internal static bool Dispatch()
         {
+            // Put threadpool thread on default context
+            ExecutionContextSwitcher.SetDefaults();
```

Comment:
Shouldn't the context be null to start with? Maybe `ExecutionContextSwitcher.CurrentIsDefault` should also check for null.